### ISSUE TITLE
Multithreaded CloudVolume Dies

### ISSUE THREAD

**Issue by william-silversmith** [seung-lab/cloud-volume#32]
@davidbuniat ran into a multithreading problem when he ran cloudvolume inside of multiple threads (it's threads all the way down 😂 ). Looks like tqdm is to blame.

tqdm issue https://github.com/tqdm/tqdm/issues/323

```Traceback (most recent call last):                                                | 0/10 [00:00<?, ?it/s]
  File "src/process_slice.py", line 22, in <module>                                | 0/8 [00:00<?, ?it/s]
    infer.process_slice(input_volume, output_volume, z=782)
  File "/cavelab/cavelab/inference/infer.py", line 85, in process_slice
    mset = self.process(input_volume, (x*step_x-self.offset, y*step_y-self.offset, z), mset)
  File "/cavelab/cavelab/inference/infer.py", line 39, in process
    inputs = self.pool.map(self.read, [(volume, (x+i*self.scale*self.width/2, y, z), (self.scale*self.width, self.scale*self.width, 1), i) for i in range(8)])
  File "/usr/local/lib/python2.7/dist-packages/multiprocess/pool.py", line 251, in map
    return self.map_async(func, iterable, chunksize).get()
  File "/usr/local/lib/python2.7/dist-packages/multiprocess/pool.py", line 567, in get
    raise self._value
AttributeError: 'tqdm' object has no attribute 'pos'
Exception AttributeError: "'tqdm' object has no attribute 'pos'" in <object repr() failed> ignored```

**Comment by william-silversmith** [comment]
Possible workaround implemented in #21

**Comment by william-silversmith** [comment]
Looks like #21 solves the problem for Davit.

**Comment by william-silversmith** [comment]
Maybe not? He ran into another problem.

**Comment by william-silversmith** [comment]
Haven't seen anyone run into this problem in a while. Closing for now, will reopen if necessary.

**Comment by william-silversmith** [comment]
This is becoming a problem for multiple people.

Example stack trace:

```test/test_cloudvolume.py::test_non_aligned_write FAILED                  [ 24%]

=================================== FAILURES ===================================
____________________________ test_non_aligned_write ____________________________

    def test_non_aligned_write():
        delete_layer()
        offset = Vec(5,7,13)
        cv, data = create_layer(size=(1024, 1024, 5, 1), offset=offset)
    
        cv[:] = np.zeros(shape=cv.shape, dtype=cv.dtype)
    
        # Write inside a single chunk
    
        onepx = Bbox( (10,200,15), (11,201,16) )
        try:
            cv[ onepx.to_slices() ] = np.ones(shape=onepx.size3(), dtype=cv.dtype)
            assert False
        except txrx.AlignmentError:
            pass
    
        cv.non_aligned_writes = True
        cv[ onepx.to_slices() ] = np.ones(shape=onepx.size3(), dtype=cv.dtype)
        answer = np.zeros(shape=cv.shape, dtype=cv.dtype)
        answer[ 5, 193, 2 ] = 1
        assert np.all(cv[:] == answer)
    
        # Write across multiple chunks
        cv[:] = np.zeros(shape=cv.shape, dtype=cv.dtype)
        cv.non_aligned_writes = True
        middle = Bbox( (512 - 10, 512 - 11, 0), (512 + 10, 512 + 11, 5) ) + offset
        cv[ middle.to_slices() ] = np.ones(shape=middle.size3(), dtype=cv.dtype)
        answer = np.zeros(shape=cv.shape, dtype=cv.dtype)
        answer[ 502:522, 501:523, : ] = 1
        assert np.all(cv[:] == answer)
    
        cv.non_aligned_writes = False
        try:
            cv[ middle.to_slices() ] = np.ones(shape=middle.size3(), dtype=cv.dtype)
            assert False
        except txrx.AlignmentError:
            pass
    
        # Big inner shell
        delete_layer()
        cv, data = create_layer(size=(1024, 1024, 5, 1), offset=offset)
        cv[:] = np.zeros(shape=cv.shape, dtype=cv.dtype)
        middle = Bbox( (512 - 150, 512 - 150, 0), (512 + 150, 512 + 150, 5) ) + offset
    
        try:
            cv[ middle.to_slices() ] = np.ones(shape=middle.size3(), dtype=cv.dtype)
            assert False
        except txrx.AlignmentError:
            pass
    
        cv.non_aligned_writes = True
>       cv[ middle.to_slices() ] = np.ones(shape=middle.size3(), dtype=cv.dtype)

test/test_cloudvolume.py:273: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cloudvolume/cloudvolume.py:836: in __setitem__
    txrx.upload_image(self, img, bbox.minpt, parallel=self.parallel)
cloudvolume/txrx.py:302: in upload_image
    download_multiple(vol, shell_chunks, fn=shade_and_upload)
cloudvolume/txrx.py:168: in download_multiple
    tq.put(dl)
cloudvolume/threaded_queue.py:255: in __exit__
    self.wait(progress=self.with_progress)
cloudvolume/threaded_queue.py:232: in wait
    self._check_errors()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <cloudvolume.threaded_queue.ThreadedQueue object at 0x7fb9b8235e10>

    def _check_errors(self):
      try:
        err = self._error_queue.get(block=False)
        self._error_queue.task_done()
        self.kill_threads()
>       raise err
E       AttributeError: 'tqdm' object has no attribute 'pos'

cloudvolume/threaded_queue.py:189: AttributeError```

**Comment by william-silversmith** [comment]
Fixed in #104 

Open tqdm PR https://github.com/tqdm/tqdm/pull/544
